app/models/submission_report.rb

```diff
@@ -0,0 +1,43 @@
+class SubmissionReport < ApplicationRecord
+  belongs_to :submission, class_name: 'TimelineEvent'
+
+  validates :status, presence: true
+
+  enum status: {
+         queued: 'queued',
+         in_progress: 'in_progress',
+         completed: 'completed'
+       }
+
+  enum conclusion: { success: 'success', failure: 'failure', error: 'error' }
+
+  validate :queued_state_is_valid
+
+  def queued_state_is_valid
+    return unless queued?
+
+    return if [started_at, completed_at, conclusion].all?(&:blank?)
+
+    errors[:status] << 'invalid queued report status'
+  end
+
+  validate :in_progress_state_is_valid
```

Comment:
Can we move the `validate` methods to the top so that it's easy to figure out the validation calls in the file?